### PR TITLE
Fix #19 Travis, linter and pre-commit hook

### LINKED ISSUE #19: Set up Travis, husky, linting
Create Travis file, and integrate husky with pre-commit hooks

### PR DESCRIPTION
**Pull request by gustafguner**
This PR sets up Travis, integrates a linter for TypeScript and attaches a pre-commit hook that automatically lints and formats the code with prettier.

### CODE REVIEW

**Comment by gustafguner** [comment]
@sashahe You can run `yarn lint` and try committing something. You will see that the pre-commit hook starts doing its business! 😄

**Comment by sashahe** [comment]
I tried this and it does not seem to work... I added a javascript file with a lot of tabs, ran yarn lint and then commited it, but it did not reformat. I tried without running yarn lint as well, and this resulted in the same behavior. Do you want me to push the test commits to this branch?

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+language: node_js
+
+node_js:
+  - '10.9.0'
```
Comment (adibbin) [inline .travis.yml]: Aren't we using 11.8.0?

**Comment by sashahe** [comment]
Yes, i typed `yarn` and packages were installed!

**Pull request merged**